scripts/releases/prebuild/compose-framework.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ * @oncall react_native
+ */
+
+const {execSync} = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+/*::
+import type { Dependency, Platform } from './types';
+*/
+
+/**
+ * Composes the final XCFramework from the artifacts in the build folder
+ */
+async function createFramework(
+  scheme /*: string */,
+  dependencies /*: $ReadOnlyArray<Dependency> */,
+  rootFolder /*: string */,
+  buildFolder /*: string */,
+) {
+  console.log('✅ Composing iOS XCFramework...');
+
+  // Get the build destination path from the prebuild step
+  const frameworksOutputFolder = path.join(buildFolder, 'Build', 'Products');
+  const frameworks = fs.readdirSync(frameworksOutputFolder);
+  console.log('Frameworks found:', frameworks.join(', '));
+
+  const frameworkPaths = frameworks.map(framework =>
+    path.join(
+      frameworksOutputFolder,
+      framework,
+      'PackageFrameworks',
+      `${scheme}.framework`,
+    ),
+  );
+
+  const output = path.join(rootFolder, `${scheme}.xcframework`);
+
+  // Check if output already exists and delete it
+  if (fs.existsSync(output)) {
+    fs.rmSync(output, {recursive: true, force: true});
+  }
+  console.log('Output path:', output);
+
+  const frameworksArgs = frameworkPaths
+    .map(framework => `-framework ${framework}`)
+    .join(' ');
+
+  const command = `xcodebuild -create-xcframework ${frameworksArgs} -output ${output}`;
+  execSync(command, {stdio: 'inherit'});
+
+  // Copy bundles into the framework
+  copyBundles(scheme, dependencies, output, frameworkPaths);
+}
+
+/**
+ * Copies the bundles in the source frameworks to the target xcframework inside the xcframework's Resources folder
+ */
+function copyBundles(
```

Comment:
as we copy the headers to the `.framework` folders, before creating the XCFrameworks, wouldn't be easier if we do the same for the resources?
Perhaps we can just reuse part fo the code for copying the headers. WDYT?

Reply:
Headers are copied automatically from the different built frameworks, while resources are not - they need to be copied manually after XCFrameworks are built. I'm 98% sure this was why I had to do it this way.